sample/src/main/java/com/google/modernstorage/sample/mediastore/MediaStoreFragment.kt

```diff
@@ -90,52 +94,60 @@ class MediaStoreFragment : Fragment() {
         binding.mediastoreType.check(R.id.type_image)
         binding.mediastoreSource.check(R.id.source_internet)
 
-        binding.addMedia.setOnClickListener {
+        binding.addMedia.setOnClickListener { view ->
             val type = when (binding.mediastoreType.checkedButtonId) {
                 R.id.type_image -> MediaType.IMAGE
                 R.id.type_video -> MediaType.VIDEO
                 else -> throw Exception("This is not supposed to happen")
             }
 
             when (binding.mediastoreSource.checkedButtonId) {
-                R.id.source_internet -> downloadMedia(type)
-                R.id.source_camera -> captureMedia(type)
-                else -> throw Exception("This is not supposed to happen")
+                R.id.source_internet -> downloadMedia(view, type)
+                R.id.source_camera -> captureMedia(view, type)
+                else -> throw IllegalArgumentException("Unsupported type: $type")
             }
         }
     }
 
-    private fun downloadMedia(type: MediaType) {
-        viewModel.setLoadingStatus(true)
-
-        when (type) {
-            MediaType.IMAGE -> {
+    private fun downloadMedia(view: View, type: MediaType) =
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewModel.setLoadingStatus(true)
 
-                viewModel.saveRandomImageFromInternet { imageUri ->
-                    viewModel.setLoadingStatus(false)
-                    viewModel.setCurrentMedia(imageUri)
+            when (type) {
+                MediaType.IMAGE -> {
+                    viewModel.saveRandomMediaFromInternet(type = FileType.IMAGE).onFailure {
+                        viewModel.setLoadingStatus(false)
+                        Snackbar.make(view, it.toString(), Snackbar.LENGTH_SHORT).show()
+                    }
+                }
+                MediaType.VIDEO -> {
+                    viewModel.saveRandomMediaFromInternet(type = FileType.VIDEO).onFailure {
+                        viewModel.setLoadingStatus(false)
+                        Snackbar.make(view, it.toString(), Snackbar.LENGTH_SHORT).show()
+                    }
                 }
-            }
-            MediaType.VIDEO -> viewModel.saveRandomVideoFromInternet { videoUri ->
-                viewModel.setLoadingStatus(false)
-                viewModel.setCurrentMedia(videoUri)
             }
         }
-    }
-
-    private fun captureMedia(type: MediaType) {
-        viewModel.setLoadingStatus(true)
 
-        viewModel.createMediaUriForCamera(type) { uri ->
-            when (type) {
-                MediaType.IMAGE -> {
-                    viewModel.saveTemporaryCameraImageUri(uri)
-                    actionTakeImage.launch(uri)
+    private fun captureMedia(view: View, type: MediaType) =
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewModel.setLoadingStatus(true)
```

Comment:
Is this something the viewModel can infer when you call `createMediaUriForCamera` and returns the result?